app/models/container_template.rb

```diff
@@ -12,4 +12,55 @@ class ContainerTemplate < ApplicationRecord
   serialize :objects, Array
 
   acts_as_miq_taggable
+
+  def instantiate(params, project = nil)
+    @ems = ext_management_system
+    @clients = {}
+    project ||= container_project.name
+
+    temp = {
+      :metadata   => {
+        :name      => name,
+        :namespace => project
+      },
+      :objects    => objects,
+      :parameters => params
+    }
+
+    processed_template = process(@ems.connect, temp)
+    create_objects(processed_template['objects'], project) if processed_template
+  end
+
+  def process(client, temp)
+    client.process_template(temp)
+  rescue KubeException => e
+    $log.error("Unexpected Exception while processing template: #{e}")
+    false
+  end
+
+  def create_objects(objects, project)
+    objects.each do |obj|
+      obj = obj.symbolize_keys
+      obj[:metadata][:namespace] = project
+      method_name = "create_#{obj[:kind].underscore}"
+      begin
+        connect_client(obj[:apiVersion], method_name).send(method_name, obj)
+      rescue KubeException => e
+        $log.error("Unexpected Exception while creating object: #{e}")
+      end
+    end
+  end
+
+  def connect_client(api_version, method_name)
```

Comment:
This feels like it should be implemented at the provider layer, not in the `container_template` model.

@blomquisg Thoughts?  Do other providers have anything like this?